### PR TITLE
some future proofing stuff

### PR DESCRIPTION
**Pull request by bradymiller**
1. Dropped support for php 7.3. (**DONE**)
2. Turned on full CI testing for php 8.1. (**DONE**)
3. Updated dev environment and CI testing to use MariaDB 10.6.  (**DONE**)
4. Updated composer and js packages.
    - Got it working with most recent `lcobucci/jwt` (https://lcobucci-jwt.readthedocs.io/en/stable/upgrading/)). (**DONE**)
    - huge amount of following during the css builds (have fixed OpenEMR codebase issues; only culprits left are @fortawesome/fontawesome-free and @ttskch/select2-bootstrap4-theme which they will need to fix themselves; note this does not break anything and just throws bunch of deprecation warnings during the build). (**DONE**)
      ```
      DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.
      Recommendation: math.div(0.5rem, 2)
      More info and automated migrator: https://sass-lang.com/d/slash-div
      ```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,65 +11,65 @@
         "source": "https://github.com/openemr/openemr"
     },
     "require": {
-        "php": ">=7.3.0",
+        "php": ">=7.4.0",
         "academe/omnipay-authorizenetapi": "3.1.2",
-        "adodb/adodb-php": "5.21.0",
+        "adodb/adodb-php": "5.21.02",
```
Comment (bradymiller) [inline composer.json:16]: oops, should be 5.21.2

**Comment by sjpadgett** [comment]
Dicom app code needs to change to update. It's on my list...

**Comment by bradymiller** [comment]
ok, this appears to be testing well, so gonna bring in.

summary:
- bye bye php 7.3
- hello php 8.1 (works on testing but lots of deprecation notices mostly in third party packages but also in openemr core)
- hello mariadb 10.6
- hello updated composer and npm third party packages (couldn't touch dwv dicom viewer since things just broke horribly when i tried to update it, so gonna stick with what works :) )
- @stephenwaite , note worked in the new version of lcobucci/jwt which is testing well
- @all, note there will be a bunch of sass build deprecation (nothing is broken, just ugly to see all the messages during build) notices in the build until @fortawesome/fontawesome-free and @ttskch/select2-bootstrap4-theme fix up their packages .

**Comment by bradymiller** [comment]
oops, meant @adunsulag in my above point on updating lcobucci/jwt 
sorry @stephenwaite , but looking forward to the congratulatory gif :)

**Comment by stephenwaite** [comment]
nice

**Comment by ivmartel** [comment]
Hi, please tell me if I can be of any help on the dwv update. I just released v0.30 that fixes the i18next update I see @bradymiller was trying to achieve.

**Pull request merged**